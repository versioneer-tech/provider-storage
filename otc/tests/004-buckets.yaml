---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-john
spec:
  bucketAccessGrants:
  - bucketName: b-john
    grantee: jane
    permission: WriteOnly
  bucketAccessRequests:
  - bucketName: b-joe
    reason: Need access
  - bucketName: b-jeff
    reason: Need access
  - bucketName: b-jane
    reason: Need access
  buckets:
  - bucketName: b-john
  principal: john
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: bucket-b-john, observe-bucketpolicy-b-john, observe-credential-john,
      and 2 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-b-john
  labels:
    crossplane.io/composite: s-john
  name: bucket-b-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: Bucket
      metadata:
        name: b-john
      spec:
        forProvider:
          bucket: b-john
          region: eu-nl
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-bucketpolicy-b-john
  labels:
    crossplane.io/composite: s-john
  name: observe-bucketpolicy-b-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-b-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-credential-john
  labels:
    crossplane.io/composite: s-john
  name: observe-credential-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-user-jane
  labels:
    crossplane.io/composite: s-john
  name: observe-user-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jane
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-john
  labels:
    crossplane.io/composite: s-john
  name: user-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: john
      spec:
        forProvider:
          enabled: true
          name: john
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
