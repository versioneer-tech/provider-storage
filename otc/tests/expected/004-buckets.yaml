# Copyright 2025, EOX (https://eox.at) and Versioneer (https://versioneer.at)
# SPDX-License-Identifier: Apache-2.0

---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-john
spec:
  bucketAccessGrants:
  - bucketName: s-john
    grantedAt: "2025-09-29T10:28:00Z"
    grantee: jane
    permission: None
  bucketAccessRequests:
  - bucketName: s-joe
    reason: Need access
    requestedAt: "2025-09-29T10:25:00Z"
  - bucketName: s-jeff
    reason: Need access
    requestedAt: "2025-09-29T10:26:00Z"
  - bucketName: s-jane
    reason: Need access
    requestedAt: "2025-09-29T10:27:00Z"
  buckets:
  - bucketName: s-john
    discoverable: true
  principal: john
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: bucket-s-john, observe-bucketpolicy-s-john, observe-credential-john,
      and 2 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-john
  labels:
    crossplane.io/composite: s-john
  name: bucket-s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: Bucket
      metadata:
        name: s-john
      spec:
        forProvider:
          bucket: s-john
          region: eu-nl
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-bucketpolicy-s-john
  labels:
    crossplane.io/composite: s-john
  name: observe-bucketpolicy-s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-s-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-credential-john
  labels:
    crossplane.io/composite: s-john
  name: observe-credential-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-user-jane
  labels:
    crossplane.io/composite: s-john
  name: observe-user-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jane
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-john
  labels:
    crossplane.io/composite: s-john
  name: user-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-john
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: john
      spec:
        forProvider:
          access_type: programmatic
          enabled: true
          name: john
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
